### DISCUSSION TITLE
Efficiency of air reading PNG pictures

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#1239] [Q&A]
When air loads a PNG with transparent pixels (3840 * 2160), the PNG picture is only 20KB, but the efficiency is also very low if 10 PNG pictures are read at the same time.

I wonder if there is a way to ignore the transparent area of the picture for rendering and then display it to the stage without using Photoshop to process PNG pictures? Because after I cut the transparent area of PNG with Photoshop, the efficiency is very high, probably because. However, the requirements of some projects are like this. The advantage is that there is no need to place the graphics and directly generate animation display.
![image](https://user-images.githubusercontent.com/91530077/136219045-83b750ab-23e8-4eb4-8c9c-ad5af61bee67.png)

**Comment by ajwfrost** [comment]
Hi

Just to check, when you say to ignore the transparent area, do you mean we should draw a solid colour instead of transparency? Which colour should be chosen?

I suspect there will be some performance hit if you're trying to alpha-blend a very large PNG on the display list, so it would perhaps be better if there are large, completely transparent areas of this, to trim them out? Or you could hook up to one of the photoshop online APIs to do this programmatically perhaps?

thanks
